docker/scale.yaml

```diff
@@ -0,0 +1,34 @@
+version: '3.7'
+
+services:
+  worker:
+    ports:
+    - '${SUPERBLOCKS_WORKER_METRICS_PORT:-9090}'
+  controller:
+    ports:
+    - '${SUPERBLOCKS_AGENT_PORT:-8020}'
+    - '${SUPERBLOCKS_WORKER_PORT:-5001}'
+  proxy:
+    # This is the NGINX config file that this image is using.
+    #
+    # worker_processes 1;
+    # error_log /dev/stdout;
+    # pid /tmp/nginx.pid;
+    # events {
+    #     worker_connections 1024;
+    # }
+    # http {
+    #     server {
+    #         listen 8020;
+    #         listen [::]:8020;
+    #         server_name _;
+    #         location / {
+    #             proxy_pass http://controller:8020;
+    #         }
+    #     }
+    # }
+    image: ${SUPERBLOCKS_DOCKER_PROXY_REPOSITORY:-ghcr.io/superblocksteam/agent-proxy}:${SUPERBLOCKS_DOCKER_PROXY_TAG:-latest}
+    depends_on:
+    - controller
+    ports:
+    - '${SUPERBLOCKS_AGENT_PORT:-8020}:${SUPERBLOCKS_AGENT_PORT:-8020}'
```

Comment:
Would having a bunch of controllers registered but otherwise not reachable by workers on other nodes affect the controller reconcile loop? I'd imagine it'll just throw a bunch of errors every time

Reply:
This compose script is agnostic of the nodes it's on since it's all on the same docker bridge network. Now if someone run 2 compose files side by side on different notes then yes the workers of one would try to connect to the workers of another but that's a side affect of the architecture. This compose file is meant to be fully encompassing and not bent to be run twice.

Reply:
Okay, let's get this in for now since it allows users to deploy a properly functioning stack in a single instance. I personally think it might still be valuable to have a separate compose file that allows them to deploy a single controller + n workers with fixed host ports on the controller so it can scaled horizontally across instances but that doesn't block this from going in.